docs/examples/example_automotive_industry_sirene.ipynb

```diff
@@ -443,36 +210,47 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 7,
-   "id": "f8e9d35e-ec32-4ed0-bc66-96ba11df091d",
+   "execution_count": 26,
+   "id": "3b089e95-4770-4ece-a59f-60171cd921a5",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# make cleaned labels\n",
-    "match_list = ['RENAULT SAS', 'ALPINE', 'BATILLY', 'MAUBEUGE CONSTRUCTION',\n",
+    "def get_carmaker_pattern(x):\n",
+    "    match_list = ['RENAULT SAS', 'ALPINE', 'BATILLY', 'MAUBEUGE CONSTRUCTION',\n",
+    "              'TOYOTA', 'STELLANTIS AUTO SAS', 'RENAULT TRUCKS']\n",
+    "    \n",
+    "    values = ['RENAULT SAS', 'RENAULT SAS', 'RENAULT SAS', 'RENAULT SAS',\n",
     "              'TOYOTA', 'STELLANTIS AUTO SAS', 'RENAULT TRUCKS']\n",
-    "other_string_list = '|'.join(['[^' + x + ']' for x in match_list])\n",
-    "\n",
-    "conditions = [gdf['denominationUniteLegale'].str.contains(x) for x in match_list]\n",
-    "conditions += [gdf['denominationUniteLegale'].str.contains(other_string_list)]\n",
-    "\n",
-    "values = ['RENAULT SAS', 'RENAULT SAS', 'RENAULT SAS', 'RENAULT SAS',\n",
-    "          'TOYOTA', 'STELLANTIS AUTO SAS', 'RENAULT TRUCKS', 'OTHER']\n",
     "\n",
-    "gdf['label'] = np.select(conditions, values)"
+    "    pattern = [values[i] for i, p in enumerate(match_list) if p in x]\n",
+    "    \n",
+    "    if len(pattern)  == 0:\n",
+    "        return \"OTHER\"\n",
+    "    else:\n",
+    "        return pattern[0]"
```

Comment:
you can use a return early pattern, here: drop the ``else`` line and just keep ``return pattern[0]``